src/octoprint/util/comm.py

```diff
@@ -583,6 +587,16 @@ def getStateString(self, state=None):
 			return "Transferring file to SD"
 		return "Unknown State (%d)" % (self._state)
 
+	def isSendingFileToSDWithSoftwareFlow(self):
+		# self._currentLine > 1 - active only after the first ok is received, if not, it will lock...
```

Comment:
Yes, this could be a problem.. the check is that to switch to software flow control after the first line is sent. If not, it will lock, if the ok is received, contine_transfer  from handle_ok will not be executed and in send_loop it will not be executed too... so it locks. I think the issue here is the order in time how the flags are set and how handle_ok and send_loop are executed in time.. any idea on how to fix this?

Reply:
I have to admit, I'm not entirely sure what the core problem is that line number check is attempting to fix. I understand it locked up if no `ok` had been received yet? But why/where?

In any case, instead of checking for a specific line number which as pointed out can change mid-print or really at any time thanks to an `M110`, maybe a global flag after initial handshake might make more sense. But that depends a bit on the original issue which sadly - as I said - I can't fully wrap my head around right now.

Any clarification would be highly appreciated 😅